### PR TITLE
Bluetooth: controller: Fixes for CIG with two CISes and separate ACLs

### PR DESCRIPTION
**Pull request by mtpr-ot**
Bluetooth: controller: Fixes for CIG with two CISes and separate ACLs

Prevent referencing inactive CISes by clearing state variables
after teardown. This was seen as necessary when having CISes on
different ACLs, and disconnecting ACLs in random order.

Signed-off-by: Morten Priess <mtpr@oticon.com>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_conn_iso.c [hunk 1]
```diff
@@ -1033,10 +1034,14 @@ static void cis_disabled_cb(void *param)
...
 				cig->lll.num_cis--;
+
+			} else if (IS_CENTRAL(cig)) {
+				/* Prevent referencing inactive CIS */
+				cis->lll.flushed = 0U;
+				cis->lll.acl_handle = LLL_HANDLE_INVALID;
 			}
 
 			/* CIS is no longer active */
-			cis->lll.active = 0U;
```
Comment (mtpr-ot) [inline subsys/bluetooth/controller/ll_sw/ull_conn_iso.c:1039]: Not needed. Always set to 0 in 'flush'

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c [hunk 2]
```diff
@@ -523,7 +523,7 @@ void ll_cis_create(uint16_t cis_handle, uint16_t acl_handle)
 	cis->lll.acl_handle = acl_handle;
 
 	/* Create access address */
-	err = util_aa_le32(cis->lll.access_addr);
+	err = util_aa_isr_le32(cis->lll.access_addr, 1U);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_central_iso.c:526]: `ll_cis_create()` executes in Thread context (HCI commands from Host). Is it incorrect to use `isr_context` as `true`.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_central_iso.c (hunk 2 above)
Comment (mtpr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_central_iso.c:526]: Yes you're right. In my workspace I had the code from before your change, where `util_aa_le32` is called in the CIS_IND preparation. When preparing the PR I didn't consider the impact of that change. Thanks.

**Review by cvinayak — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_conn_iso.c (hunk 1 above)
Comment (niag-Oticon) [reply subsys/bluetooth/controller/ll_sw/ull_conn_iso.c:1039]: I think I missed this point. Where is `cis->lll.active` cleared on flush?
The rest looks good to me.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_conn_iso.c:1039]: Good question!

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_conn_iso.c (hunk 1 above)
Comment (mtpr-ot) [reply subsys/bluetooth/controller/ll_sw/ull_conn_iso.c:1039]: This is done in `cis_tx_lll_flush`, which in my opinion is the correct place.

**Review by niag-Oticon — APPROVED** [review]
LGTM.

**Comment by mtpr-ot** [comment]
@cvinayak, can I get your approval? - question answered.

**Review by cvinayak — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**